LibraryFeature: Rename getChildModel to sidebarModel and make it const

Holzhaus opened the pull request.

Holzhaus committed:
LibraryFeature: Rename getChildModel to sidebarModel and make it const

uklotzde commented on src/library/autodj/autodjfeature.h:
```diff
@@ -62,7 +62,7 @@ class AutoDJFeature : public LibraryFeature {
     int m_iAutoDJPlaylistId;
     AutoDJProcessor* m_pAutoDJProcessor;
-    TreeItemModel m_childModel;
+    TreeItemModel* m_pSidebarModel;
```
For this purpose we have `parented_ptr`

Holzhaus replied:
Doesn't work. Although `QAbstractItemModel` inherits from QObject, I get:

```
/home/jan/Projects/mixxx/src/util/parented_ptr.h: In instantiation of ‘parented_ptr<T>::~parented_ptr() [with T = TreeItemModel]’:
/home/jan/Projects/mixxx/src/library/itunes/itunesfeature.cpp:67:55:   required from here
/home/jan/Projects/mixxx/src/util/parented_ptr.h:31:45: error: no matching function for call to ‘TreeItemModel::parent()’
   31 |         DEBUG_ASSERT(!m_ptr || m_ptr->parent());
      |                                ~~~~~~~~~~~~~^~
```

uklotzde replied:
Caused by conflicting method names in the Qt class hierarchy and a deficiency of `parented_ptr`:

```
DEBUG_ASSERT(!m_ptr || static_cast<const QObject*>(m_ptr)->parent());
```

Please fix it and try again.

uklotzde commented on src/library/banshee/bansheefeature.cpp:
```diff
@@ -18,6 +18,7 @@ QString BansheeFeature::m_databaseFile;
 
 BansheeFeature::BansheeFeature(Library* pLibrary, UserSettingsPointer pConfig)
         : BaseExternalLibraryFeature(pLibrary, pConfig),
+          m_pSidebarModel(new TreeItemModel(this)),
```
This becomes `make_parented<TreeItemModel>(this)`

Holzhaus committed:
parented_ptr: Cast to QObject to fix parent method signature mismatch

Classes inheriting from QAbstractItemModel are QObjects, but also have a
`parent()` method that takes a QModelIndex. In these cases, an error
like this may be thrown:

    src/util/parented_ptr.h: In instantiation of ‘parented_ptr<T>::~parented_ptr() [with T = TreeItemModel]’:
    src/library/itunes/itunesfeature.cpp:67:55:   required from here
    src/util/parented_ptr.h:31:45: error: no matching function for call to ‘TreeItemModel::parent()’
       31 |         DEBUG_ASSERT(!m_ptr || m_ptr->parent());
          |                                ~~~~~~~~~~~~~^~

This fixes the issue by casting to QObject first.

Holzhaus committed:
library: Use parented_ptr for feature sidebar model instances

uklotzde reviewed (approved):
Thank you! LGTM
